platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy

```diff
@@ -263,7 +239,33 @@ class SoftwareTypeDeclarationIntegrationTest extends AbstractIntegrationSpec imp
 
         then:
         failure.assertHasCause("Failed to apply plugin 'com.example.test-software-type'.")
-        failure.assertHasCause("A plugin with type 'org.gradle.test.SoftwareTypeImplPlugin' was registered as a software type plugin, but it does not expose any software types. Software type plugins must expose software types via properties with the @SoftwareType annotation.")
+        failure.assertHasCause("A problem was found with the SoftwareTypeImplPlugin plugin.")
+        failure.assertHasCause("Type 'org.gradle.test.SoftwareTypeImplPlugin' is registered as a software type plugin but does not expose a software type.")
+    }
+
+    def 'sensible error when a software type plugin is registered that exposes multiple software types'() {
```

Comment:
💭 Probably also should have a case ensuring you can register multiple software types in multiple plugins using single RegistersSoftwareTypes plugin.

Reply:
There already tests for this [here](https://github.com/gradle/gradle/blob/e60c047483af8dd39fc28715556282dbe1f24dd7/platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy#L110) and [here](https://github.com/gradle/gradle/blob/e60c047483af8dd39fc28715556282dbe1f24dd7/platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/settings/SoftwareTypeDeclarationIntegrationTest.groovy#L138).